### DISCUSSION TITLE
socket doen't work in express in routes

### DISCUSSION THREAD

**Discussion by jensysantana** [socketio/socket.io#4001] [Q&A]
app.get('/', async (req, res) => {

            io.on('connection', (socket) => {
                socket.on('disconnect', () => {
                       
                 });
                socket.emit('messagex', ({ name: 'anyname' }));
            });
            
});

**Comment by darrachequesne** [comment]
You shouldn't declare you `io.on('connection', ...)` in your route:

```js
app.get('/', async (req, res) => {
  // ...
});

io.on('connection', (socket) => {
  socket.on('disconnect', () => {
         
  });
  
  socket.emit('messagex', ({ name: 'anyname' }));
});
```
